### PR TITLE
allow to toggle "Organize save files" at runtime

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

File: src/d_main.c [hunk 1]
```diff
@@ -1759,14 +1708,121 @@ static boolean CheckHaveSSG (void)
...
+            free(oldsavegame);
+
+            M_MakeDirectory(basesavegame);
+
+            oldsavegame = basesavegame;
+            basesavegame = M_StringJoin(oldsavegame, DIR_SEPARATOR_S,
+            M_BaseName(wadname));
+            free(oldsavegame);
+
+            M_MakeDirectory(basesavegame);
```
Comment (rfomin) [inline src/d_main.c:1793]: What if we create directory only when user saves game? I have hundreds of empty directories in `savegames` (my case is unusual thought).

File: src/d_main.c [hunk 2]
```diff
@@ -1710,22 +1710,20 @@ static boolean CheckHaveSSG (void)
 
 static int mainwadfile;
 
+#define SET_DIR(a, b) \
```
Comment (rfomin) [inline src/d_main.c:1713]: I think we don't need the `if (a) { free(a); }` construction, I remember adding it a long time ago when I removed the `PATH_MAX` limit. Now I'm pretty sure that we don't allocate these strings anywhere else, so we can simplify this code.

Comment (fabiangreffrath) [reply src/d_main.c:1713]: The point is that it can be run from the menu multiple times per game now.

Comment (rfomin) [reply src/d_main.c:1713]: Okay, got it.

**Review by rfomin — APPROVED** [review]

**Pull request merged**